### PR TITLE
Ability to terminate a node at given height

### PR DESCRIPTION
**Pull request by lgalabru**
With this changes, neon nodes will exit, and keep exiting if the height of the burnchain they're following reached a given value.
This should be deployed when as soon as the CRON resetting the burnchain is deployed.
Per devops demand, the NEON_GENESIS_TIMESTAMP can now override `neon.genesis_timestamp` (from the toml config) used by bitcoin_neon_controller via an ENV VAR.

### CODE REVIEW

**Comment by diwakergupta** [comment]
@lgalabru forgot to add reviewers? ;-)

File: testnet/bitcoin-neon-controller/src/main.rs [hunk 1]
```diff
@@ -47,19 +47,31 @@ async fn main() -> http_types::Result<()> {
             Err(err) => err.duration(),
         }.as_secs() as u64;
 
-        let time_since_genesis = now - config.neon.genesis_timestamp;
-        
+        let genesis_timestamp = match std::env::var("NEON_GENESIS_TIMESTAMP") {
+            Ok(val) =>  match val.parse::<u64>() {
+                Ok(val) => val,
+                _ => config.neon.genesis_timestamp,
```
Comment (jcnelson) [inline testnet/bitcoin-neon-controller/src/main.rs:53]: Should emit a warning here if the parse fails.

File: testnet/bitcoin-neon-controller/src/main.rs [hunk 2]
```diff
@@ -47,19 +47,31 @@ async fn main() -> http_types::Result<()> {
...
+            },
+            _ => config.neon.genesis_timestamp
```
Comment (kantai) [inline testnet/bitcoin-neon-controller/src/main.rs:55]: Would it make sense to default to something like `now` for the genesis_timestamp? That way this controller could be used in local setups a little more easily.

Comment (jcnelson) [reply testnet/bitcoin-neon-controller/src/main.rs:55]: Seconded.  I hit this problem while testing.  Having to set a timestamp in the config file every time I wanted to spin up a fresh test environment was annoying.

Comment (lgalabru) [reply testnet/bitcoin-neon-controller/src/main.rs:55]: Fine by me, we'll just have to be cautious on the master node, and ensure that the env var is set (which should be fine with our move to k8s)

Comment (lgalabru) [reply testnet/bitcoin-neon-controller/src/main.rs]: I'm a bit ambivalent about the approach of being destructive and resetting by default, how do you guys feel about https://github.com/blockstack/stacks-blockchain/pull/1577/commits/f2dedf7dbc18e138d0cc1aadfc1c64c9b3cc2f34?

File: testnet/bitcoin-neon-controller/src/main.rs (hunk 1 above)
Comment (lgalabru) [reply testnet/bitcoin-neon-controller/src/main.rs]: Addressed with https://github.com/blockstack/stacks-blockchain/commit/f2dedf7dbc18e138d0cc1aadfc1c64c9b3cc2f34

File: testnet/bitcoin-neon-controller/src/main.rs (hunk 2 above)
Comment (kantai) [reply testnet/bitcoin-neon-controller/src/main.rs]: That works for me

**Comment by lgalabru** [comment]
This feature was disabled for neon (to avoid compatibility issues with nodes running out there), and enabled it for argon. Merging now.

**Pull request merged**